src/lua/LuaMusic.cpp

```diff
@@ -194,10 +194,11 @@ static int l_music_get_song_list(lua_State *l)
 {
 	using std::string;
 	using std::vector;
-	const vector<string> vec = Pi::GetMusicPlayer().GetSongList();
+	vector<string> songList;
+	Pi::GetMusicPlayer().GetSongList(songList);
```

Comment:
Return-value optimization should be automatically applied here - there's no need to change the API. Returning a vector by value is intentional and desired in this case.

Reply:
ok I'll change this back